src/blocks/Transaction.vue

```diff
@@ -281,586 +229,501 @@ export default {
-       * Just a check to keep button disabled and avoid showing a message.
-       */
-      if (!val || !parseFloat(val) || !this.chosenToken) {
-        this.setMainError("");
-        return false;
+    async getAccountActivationFee(): Promise<void> {
+      if (!this.feeToken && !this.ownAccountUnlocked) {
+        return;
       }
-
-      let inputAmount = null;
-
-      /**
-       * If validated too early
-       */
+      this.activateAccountFeeLoading = true;
+      const syncWallet = walletData.get().syncWallet;
+      const syncProvider = walletData.get().syncProvider;
       try {
-        inputAmount = utils.parseToken(this.chosenToken.symbol, val);
+        await this.$accessor.wallet.restoreProviderConnection();
+        const foundFee = await syncProvider?.getTransactionFee(
+          {
+            ChangePubKey: syncWallet?.ethSignerType?.verificationMethod === "ERC-1271" ? "Onchain" : "ECDSA",
+          },
+          syncWallet?.address() || "",
+          this.feeToken.symbol,
+        );
+        this.activateAccountFee = foundFee!.totalFee.toString();
       } catch (error) {
-        let errorInfo = `Amount processing error. Common reason behind it — inaccurate amount. Try again paying attention to the decimal amount number format — it should help`;
-        if (error.message && error.message.search("fractional component exceeds decimals") !== -1) {
-          errorInfo = `Precision exceeded: ${this.chosenToken.symbol} doesn't support that many decimal digits`;
-        }
-        this.pushDigitsErrorValue(errorInfo);
-        return false;
+        await this.$accessor.toaster.error(error.message ? error.message : "Error while receiving an unlock fee");
       }
-      this.pushDigitsErrorValue();
-
-      if (inputAmount.lte(0)) {
-        this.hasValidAmount = false;
-        return false;
+      this.activateAccountFeeLoading = false;
+    },
+    async activateAccount(): Promise<string> {
+      if (this.activateAccountFee === undefined || !this.feeToken) {
+        return "";
       }
+      this.error = "";
+      this.loading = true;
+      try {
+        this.clearTransactionInfo();
+        const syncWallet = walletData.get().syncWallet;
+        await this.$accessor.wallet.restoreProviderConnection();
+        this.tip = "Confirm the transaction to unlock this account";
+
+        if (syncWallet?.ethSignerType?.verificationMethod === "ERC-1271") {
+          const isOnchainAuthSigningKeySet = await syncWallet!.isOnchainAuthSigningKeySet();
+          if (!isOnchainAuthSigningKeySet) {
+            const onchainAuthTransaction = await syncWallet!.onchainAuthSigningKey();
+            await onchainAuthTransaction?.wait();
+          }
 
-      this.inputTotalSum = val;
-      this.inputTotalSumBigNumber = inputAmount;
+          const isSigningKeySet = await syncWallet!.isSigningKeySet();
+          if (!isSigningKeySet) {
+            const changePubkey = await syncWallet?.setSigningKey({
```

Comment:
The arguments for CPK have changed in the new sdk. 

Please, look at this PR: https://github.com/matter-labs/zksync-docs/pull/69

Reply:
@StanislavBreadless resolved